bin/git-cp

```diff
@@ -32,9 +32,36 @@ elif [[ "$CURRENT_FILENAME" == "" ]]; then
     usage
     exit 10  # Missing arguments CURRENT_FILENAME
 else
+    if [ -e "$DESTINATION_FILENAME"  ]; then
+        echo 1>&2 "$DESTINATION_FILENAME already exists."
+        echo 1>&2 "Make sure to remove the destination first."
+        exit 40
+    fi
+
     echo "Coping $CURRENT_FILENAME into $DESTINATION_FILENAME"
-    cp -r "${CURRENT_FILENAME}" "${DESTINATION_FILENAME}"
 
-    git add "$DESTINATION_FILENAME"
-    git commit -m "Duplicate ${CURRENT_FILENAME} history in ${DESTINATION_FILENAME} history." -n
+    INTERMEDIATE_FILENAME="${CURRENT_FILENAME//\//__}-move-to-${DESTINATION_FILENAME//\//__}"
+
+    # We keep the existing file on the side in a commit
+    git mv "${CURRENT_FILENAME}" "${INTERMEDIATE_FILENAME}"
+    git commit -a -nm "Keep $CURRENT_FILENAME"
```

Comment:
Why do we need to use `-a` after `git mv`?

Reply:
It is the test to avoid interruption because the original implementation was interrupted by conflict always. At first, it confused me and I thought something missing maybe, so the option `-a` was added. The option `-a` is useless for the current implementation and I forget to remove it. Fixed already.